tests/extract-costfn.egg

```diff
@@ -0,0 +1,12 @@
+(datatype E
+  (Foo String))
+
+(union (Foo "x") (Foo "y"))
+(union (Foo "y") (Foo "z"))
+
+(unstable-cost (Foo "x") 17)
+(unstable-cost (Foo "y") 11)
+(unstable-cost (Foo "z") 15)
+
+(extract (Foo "y"))
```

Comment:
This test doesn't check that we get a particular result- perhaps it should be a rust test that can?

Reply:
Added this as a rust test